homeassistant/components/zha/config_flow.py

```diff
@@ -1,80 +1,126 @@
 """Config flow for ZHA."""
-import asyncio
-from collections import OrderedDict
 import os
+from typing import Any, Dict, Optional
 
+import serial.tools.list_ports
 import voluptuous as vol
+from zigpy.config import CONF_DEVICE, CONF_DEVICE_PATH
 
 from homeassistant import config_entries
 
-from .core.const import (
+from .core.const import (  # pylint:disable=unused-import
     CONF_RADIO_TYPE,
-    CONF_USB_PATH,
     CONTROLLER,
-    DEFAULT_BAUDRATE,
-    DEFAULT_DATABASE_NAME,
     DOMAIN,
-    ZHA_GW_RADIO,
     RadioType,
 )
 from .core.registries import RADIO_TYPES
 
+CONF_MANUAL_PATH = "Enter Manually"
 
-@config_entries.HANDLERS.register(DOMAIN)
-class ZhaFlowHandler(config_entries.ConfigFlow):
+
+class ZhaFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
     """Handle a config flow."""
 
-    VERSION = 1
+    VERSION = 2
     CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_PUSH
 
+    def __init__(self):
+        """Initialize flow instance."""
+        self._device_path = None
+        self._radio_type = None
+
     async def async_step_user(self, user_input=None):
         """Handle a zha config flow start."""
         if self._async_current_entries():
             return self.async_abort(reason="single_instance_allowed")
 
-        errors = {}
+        ports = serial.tools.list_ports.comports(include_links=False)
```

Comment:
It does do `glob.glob()` so I guess it does do IO. I'll add it into executor job, just thought since it is all against udev it doesn't do IO to physical or network block devices.

Reply:
Done in executor now.